### PR TITLE
IAUX icons monorepo

### PR DESCRIPTION
**Pull request by shaneriley**
The package for ia-icons within IAUX has been moved here as one of a number of packages managed in a monorepo setup using Lerna. Using this pattern, each icon will be published as its own package under the `@internetarchive` group, prefixed with 'icon-'. Note that the individual icons are not components themselves, rather they are lit-html template objects. These are meant to be used within a lit-element component, like what is demonstrated in the ia-icons package.

A caveat is added to the README, but as a reminder Lerna will install an instance of lit-element for EVERY package in this repo, and when ia-icons imports all of the icons, duplicate instances of lit-element are included which causes rendering issues. It is recommended that you go to the ia-icons package directory and run `npm i` there only, as this is significantly faster than bootstrapping via Lerna and then running `npm dedupe` within the ia-icons package.

### CODE REVIEW

File: packages/ia-icons/package.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    "test": "karma start --coverage",
+    "test:watch": "karma start --auto-watch=true --single-run=false",
+    "test:update-snapshots": "karma start --update-snapshots",
+    "test:prune-snapshots": "karma start --prune-snapshots",
+    "test:compatibility": "karma start --compatibility all --coverage",
+    "test:compatibility:watch": "karma start --compatibility all --auto-watch=true --single-run=false",
+    "test:bs": "karma start karma.bs.config.js --compatibility all --coverage"
+  },
+  "dependencies": {
+    "@internetarchive/icon-advance": "0.0.1-alpha.13+91b69ae",
```
Comment (jbuckner) [inline packages/ia-icons/package.json:28]: Are you going to bump these to non-alpha versions?

File: packages/ia-icons/symlink_packages.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const path = require('path');
+const fs = require('fs');
+const { spawn } = require('child_process');
+
+const icon_packages = fs.readdirSync(path.join(__dirname, '..'), { withFileTypes: true })
```
Comment (jbuckner) [inline packages/ia-icons/symlink_packages.js:5]: Are you using this file? I didn't see it getting called.

File: packages/icon-advance/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { html } from 'lit-element';
```
Comment (jbuckner) [inline packages/icon-advance/index.js:1]: You can actually do `import { html } from 'lit-html'` instead of `lit-element`. It's a smaller dependency.

Comment (jbuckner) [reply packages/icon-advance/index.js:1]: Test

File: packages/ia-icons/symlink_packages.js (hunk 2 above)
Comment (shaneriley) [reply packages/ia-icons/symlink_packages.js:5]: Ah, no. That was a prior attempt to skipping the `npm dedupe` in favor of symlinking, but that didn't work.

File: packages/ia-icons/package.json (hunk 1 above)
Comment (shaneriley) [reply packages/ia-icons/package.json:28]: Yeah, before my merges I always add a commit to bump version number. With topnav, there were times when I had 2 or 3 PRs open, and I didn't know which would merge in first and receive the next version bump.

**Comment by jbuckner** [comment]
Something I just remembered.. lerna has a [`--hoist`](https://github.com/lerna/lerna/blob/master/doc/hoist.md) option that I think pulls up common dependencies when bootstrapping. Did you try that? I'm curious if that would also solve the duplication.

**Comment by shaneriley** [comment]
Just tried it. I get 404s for packages when I run the dev server.

File: packages/icon-advance/index.js (hunk 3 above)
Comment (shaneriley) [reply packages/icon-advance/index.js:1]: Converted them all to lit-html instead and got the same issue. The core of the problem is in lit-html, so with either package you need to do the same deduping.

Comment (jbuckner) [reply packages/icon-advance/index.js:1]: Ah okay.

**Pull request merged**